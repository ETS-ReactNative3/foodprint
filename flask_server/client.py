#!/usr/bin/env python3

import requests

address = 'http://127.0.0.1:5000'
url = address + '/upload/'
content_type = 'image/jpg'

headers = {'content-type': content_type, 'h': 100, 'w': 100}

content = {'image': open('apples.jpg', 'rb')}
print(content)
response = requests.post(url, files=content)
print(response)
