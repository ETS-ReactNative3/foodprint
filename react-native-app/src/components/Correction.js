import { Image, View, ActivityIndicator } from 'react-native';
import { Input, Text } from 'react-native-elements';
import React, { useState, useEffect } from 'react';

import { gql } from 'apollo-boost';
import { useMutation } from '@apollo/react-hooks';
import { StyleSheet } from 'react-native';
import { ScrollView } from 'react-native-gesture-handler';
import { widthPercentageToDP as percentageWidth, heightPercentageToDP as percentageHeight } from 'react-native-responsive-screen';


// GraphQL schema for correction mutation
const POST_CORRECTION_MUTATION = gql`
mutation PostCorrectionMutation($name: String!) {
  postCorrection(name: $name) {
    name
    carbonFootprintPerKg
  }
}
`;

const Correction = ({ route, navigation }) => {
  const { meal, setMeal } = route.params;
  const [correctedName, setCorrectedName] = useState('');
  const [postCorrection, { loading: correctionLoading, error: correctionError, data: correctionData }] = useMutation(POST_CORRECTION_MUTATION);

  // Handle correction from input field
  const handleCorrection = async () => {
    try {
      await postCorrection({ variables: { name: correctedName } });
    } catch (err) {
      console.warn({ err });
    }
  };

  useEffect(() => {
    if (!correctionError) {
      return;
    }
    // TODO show error to user
    console.warn('Error in correction request');
  }, [correctionError]);

  // Respond to changes in correction data (following correction)
  useEffect(() => {
    if (correctionData && correctionData.postCorrection.carbonFootprintPerKg) {
      setMeal({
        ...meal,
        score: correctionData.postCorrection.carbonFootprintPerKg,
        description: correctedName,
      });
      navigation.goBack();
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [correctionData]);

  return correctionLoading ?
    <View style={styles.loading}>
      <ActivityIndicator />
    </View > : (
      <View style={styles.container}>
        <ScrollView>
          <View style={styles.titleContainer}>
            <Text style={styles.title}>We could not find your item.</Text>
          </View>
          <View style={styles.imageContainer}>
            <Image
              style={styles.image}
              source={{ uri: 'https://mir-s3-cdn-cf.behance.net/project_modules/disp/fc4a1059120725.5a15c9fa08f78.gif' }}
            />
          </View>
          <View style={styles.inputContainer}>
            <Input
              placeholder="Cucumber"
              onChangeText={value => setCorrectedName(value.toLowerCase())}
              onSubmitEditing={handleCorrection}
            />
          </View>
          <View style={styles.subtitleContainer}>
            <Text style={styles.subtitle}>Let us know what it was, so we can improve our app.</Text>
          </View>
        </ScrollView>
      </View >
    );
};

const styles = StyleSheet.create({
  loading: {
    height: '100%',
    justifyContent: 'center',
  },
  container: { flex: 1, justifyContent: 'center' },
  titleContainer: { flex: 2, justifyContent: 'center', marginTop: percentageHeight('10%') },
  title: { textAlign: 'center', fontSize: percentageWidth('7%') },
  imageContainer: { flex: 2.5, alignItems: 'center', justifyContent: 'center', marginTop: percentageHeight('5%') },
  image: { height: percentageWidth('30%'), width: percentageWidth('30%') },
  subtitleContainer: { flex: 1, justifyContent: 'center' },
  subtitle: { fontSize: percentageWidth('5%'), textAlign: 'center', marginHorizontal: percentageWidth('5%') },
  inputContainer: { flex: 1.5, margin: percentageWidth('5%') },
});

export default Correction;
