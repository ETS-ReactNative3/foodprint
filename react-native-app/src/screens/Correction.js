import {Image, KeyboardAvoidingView, View} from 'react-native';
import { Input, Text, Button } from 'react-native-elements';
import React, { useState, useEffect } from 'react';
import LottieView from 'lottie-react-native';
import { gql } from 'apollo-boost';
import { useMutation } from '@apollo/react-hooks';
import { StyleSheet } from 'react-native';
import { widthPercentageToDP as percentageWidth, heightPercentageToDP as percentageHeight } from 'react-native-responsive-screen';
import Snackbar from 'react-native-snackbar';

// GraphQL schema for correction mutation
const POST_CORRECTION_MUTATION = gql`
mutation PostCorrectionMutation($name: String!) {
  postCorrection(name: $name) {
    name
    carbonFootprintPerKg
  }
}
`;

const Correction = ({ navigation }) => {
  const [loading, setLoading] = useState(false);
  const [correctedName, setCorrectedName] = useState('');
  const [postCorrection, { loading: correctionLoading, error: correctionError, data: correctionData }] = useMutation(POST_CORRECTION_MUTATION);

  const postCorrectionHandler = async () => {
    if (correctedName.length === 0) {
      return Snackbar.show({
        text: "If you don't know what it is, give your closest match.",
        duration: Snackbar.LENGTH_SHORT,
      });
    }
    setLoading(true);
    await postCorrection({ variables: { name: correctedName } });
  };

  useEffect(() => {
    if (correctionError) {
      setLoading(false);
      Snackbar.show({
        text: 'We couldn\'t save that to your history.',
        duration: Snackbar.LENGTH_INDEFINITE,
        action: {
          text: 'RETRY',
          textColor: 'red',
          onPress: postCorrectionHandler,
        },
      });
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [correctionError]);

  // Respond to changes in correction data (following correction)
  useEffect(() => {
    if (correctionData && correctionData.postCorrection.carbonFootprintPerKg) {
      navigation.push('Feedback', {
        meal: {
          score: correctionData.postCorrection.carbonFootprintPerKg,
          description: correctedName,
          item: correctionData.postCorrection.name, // Item returned from back-end
        },
      });
      setLoading(false);
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [correctionData]);

  return correctionLoading || loading ?
    <View style={styles.loading}>
      <LottieView source={require('../animations/18473-flying-avocado.json')} autoPlay loop />
    </View > : (
      <KeyboardAvoidingView behavior="padding" style={styles.container}>
          <View style={styles.titleContainer}>
            <Text style={styles.title}>Sorry, we could not find your item.</Text>
          </View>
          <View style={styles.imageContainer}>
            <Image
              style={styles.image}
              source={{ uri: 'https://mir-s3-cdn-cf.behance.net/project_modules/disp/fc4a1059120725.5a15c9fa08f78.gif' }}
            />
          </View>
          <View style={styles.inputContainer}>
            <Input
              placeholder="Cucumber"
              onChangeText={value => setCorrectedName(value.toLowerCase())}
              onSubmitEditing={postCorrectionHandler}
            />
          </View>
          <View style={styles.subtitleContainer}>
            <Text style={styles.subtitle}>Let us know what it was, so we can improve our app.</Text>
          </View>
          <Button
            title="SUBMIT"
            onPress={postCorrectionHandler}
            buttonStyle={styles.button}
            containerStyle={styles.buttonContainer}
            titleStyle={styles.buttonTitle}
          />
      </KeyboardAvoidingView >
    );
};

const styles = StyleSheet.create({
  loading: {
    height: '100%',
    justifyContent: 'center',
  },
  container: { flex:1, marginHorizontal:percentageWidth('5%'), justifyContent: 'center', alignSelf:'center' },
  titleContainer: { height:percentageHeight('10%'), justifyContent: 'center', alignItems:'center' },
  title: { textAlign: 'center', fontSize: percentageWidth('7%') },
  imageContainer: { height:percentageHeight('30%'), alignItems: 'center', justifyContent: 'center' },
  image: { height: percentageWidth('50%'), width: percentageWidth('50%') },
  subtitleContainer: { height:percentageHeight('7%'), justifyContent: 'center' },
  subtitle: { fontSize: percentageWidth('5%'), textAlign: 'center' },
  inputContainer: { height:percentageHeight('10%'), alignItems:'center', justifyContent:'center' },
  button: { backgroundColor: 'green', width: percentageWidth('30%'), height: 45 },
  buttonContainer: { height:percentageHeight('10%'), alignItems: 'center', marginTop:percentageHeight('2%'), paddingBottom:percentageHeight('20%') },
});

export default Correction;
