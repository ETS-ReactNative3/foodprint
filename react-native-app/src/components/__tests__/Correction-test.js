import React from 'react';
import Correction from '../Correction';

import renderer from 'react-test-renderer';
import { useMutation } from '@apollo/client';

jest.mock('@apollo/client');

// Check current component against its snapshot
test('Correction matches previous snapshot', () => {

  const mockedRoute = {
    params: {
      meal: {},
    },

  };
  useMutation.mockImplementation(() => [null, { loading: null, error: null, data: null }]);
  const tree = renderer.create(<Correction route={mockedRoute} isVisible={true} />).toJSON();
  expect(tree).toMatchSnapshot();
});
