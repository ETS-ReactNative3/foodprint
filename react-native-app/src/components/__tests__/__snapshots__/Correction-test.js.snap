// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Correction matches previous snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "flex": 2,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 35,
          "textAlign": "center",
        }
      }
      theme={
        Object {
          "colors": Object {
            "disabled": "hsl(208, 8%, 90%)",
            "divider": "#bcbbc1",
            "error": "#ff190c",
            "grey0": "#393e42",
            "grey1": "#43484d",
            "grey2": "#5e6977",
            "grey3": "#86939e",
            "grey4": "#bdc6cf",
            "grey5": "#e1e8ee",
            "greyOutline": "#bbb",
            "platform": Object {
              "android": Object {
                "error": "#f44336",
                "primary": "#2196f3",
                "secondary": "#9C27B0",
                "success": "#4caf50",
                "warning": "#ffeb3b",
              },
              "ios": Object {
                "error": "#ff3b30",
                "primary": "#007aff",
                "secondary": "#5856d6",
                "success": "#4cd964",
                "warning": "#ffcc00",
              },
            },
            "primary": "#2089dc",
            "searchBg": "#303337",
            "secondary": "#8F0CE8",
            "success": "#52c41a",
            "warning": "#faad14",
          },
        }
      }
    >
      We're sorry we couldn't find your item...
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 2.5,
        "justifyContent": "center",
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "https://mir-s3-cdn-cf.behance.net/project_modules/disp/fc4a1059120725.5a15c9fa08f78.gif",
        }
      }
      style={
        Object {
          "height": 200,
          "width": 200,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 22,
          "marginHorizontal": 20,
          "textAlign": "center",
        }
      }
      theme={
        Object {
          "colors": Object {
            "disabled": "hsl(208, 8%, 90%)",
            "divider": "#bcbbc1",
            "error": "#ff190c",
            "grey0": "#393e42",
            "grey1": "#43484d",
            "grey2": "#5e6977",
            "grey3": "#86939e",
            "grey4": "#bdc6cf",
            "grey5": "#e1e8ee",
            "greyOutline": "#bbb",
            "platform": Object {
              "android": Object {
                "error": "#f44336",
                "primary": "#2196f3",
                "secondary": "#9C27B0",
                "success": "#4caf50",
                "warning": "#ffeb3b",
              },
              "ios": Object {
                "error": "#ff3b30",
                "primary": "#007aff",
                "secondary": "#5856d6",
                "success": "#4cd964",
                "warning": "#ffcc00",
              },
            },
            "primary": "#2089dc",
            "searchBg": "#303337",
            "secondary": "#8F0CE8",
            "success": "#52c41a",
            "warning": "#faad14",
          },
        }
      }
    >
      Let us know what it was, so we can improve our app:
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1.5,
      }
    }
  >
    <View
      style={
        Object {
          "margin": 20,
        }
      }
    >
      <View
        style={
          Object {
            "paddingHorizontal": 10,
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "borderBottomWidth": 1,
              "borderColor": "#86939e",
              "flexDirection": "row",
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            editable={true}
            onChangeText={[Function]}
            onSubmitEditing={[Function]}
            placeholder="e.g. Cucumber"
            rejectResponderTermination={true}
            style={
              Object {
                "alignSelf": "center",
                "color": "black",
                "flex": 1,
                "fontSize": 18,
                "minHeight": 40,
              }
            }
            testID="RNE__Input__text-input"
            underlineColorAndroid="transparent"
          />
        </View>
      </View>
    </View>
  </View>
</View>
`;
